src/features/resources/resourcesSlice.ts

```diff
@@ -0,0 +1,611 @@
+        const types = state.searchTypeResults?.length ? state.searchTypeResults : state.allTypes;
+        const index = (state.currentTypePage - 1) * state.typeResultsAmount;
+        state.currentTypes = types.slice(index - state.typeResultsAmount, index);
+        state.currentTypePage -= 1;
+      }
+    },
+    specificPageSync: (state, action: PayloadAction<SpecificPagePayload>) => {
+      const { payload } = action;
+      const { page, resourceType } = payload;
+      if (isPokemon(resourceType)) {
+        const pokemons = state.searchPokemonResults?.length
+          ? state.searchPokemonResults
+          : state.allPokemons;
+        const index = page * state.pokemonResultsAmount;
+        state.currentPokemons = pokemons.slice(index - state.pokemonResultsAmount, index);
+        state.currentPokemonPage = page;
+      }
+      if (isMove(resourceType)) {
+        const moves = state.searchMoveResults?.length ? state.searchMoveResults : state.allMoves;
+        const index = page * state.moveResultsAmount;
+        state.currentMoves = moves.slice(index - state.moveResultsAmount, index);
+        state.currentMovePage = page;
+      }
+      if (isType(resourceType)) {
+        const types = state.searchTypeResults?.length ? state.searchTypeResults : state.allTypes;
+        const index = page * state.typeResultsAmount;
+        state.currentTypes = types.slice(index - state.typeResultsAmount, index);
+        state.currentTypePage = page;
+      }
+    },
+    searchResultsSync: (state, action: PayloadAction<SearchResultsPayload>) => {
+      const { payload } = action;
+      const { inputValue } = payload;
+      const searchedTypes = sortType(state.allTypes)
+        .byId()
+        .filter((item) => item.name.includes(inputValue));
+      const searchedPokemons = sortPokemon(state.allPokemons)
+        .byId()
+        .filter((item) => item.name.includes(inputValue));
+      const searchedMoves = sortMove(state.allMoves)
+        .byId()
+        .filter((item) => item.name.includes(inputValue));
+      state.currentPokemons = searchedPokemons.slice(0, state.pokemonResultsAmount);
+      state.currentMoves = searchedMoves.slice(0, state.moveResultsAmount);
+      state.currentTypes = searchedTypes.slice(0, state.typeResultsAmount);
+      state.searchPokemonResults = searchedPokemons;
+      state.searchMoveResults = searchedMoves;
+      state.searchTypeResults = searchedTypes;
+      state.pokemonSorting = PokemonSorting.none;
+      state.moveSorting = MoveSorting.none;
+      state.typeSorting = TypeSorting.none;
+      state.currentPokemonPage = 1;
+      state.currentMovePage = 1;
+      state.currentTypePage = 1;
+    },
+  },
+  extraReducers: (builder) => {
+    builder.addCase(setParameterizedDataAsync.pending, (state) => {
+      state.status = Status.loading;
+    }),
+      builder.addCase(setParameterizedDataAsync.fulfilled, (state, action) => {
+        const { payload } = action;
+        if (payload) {
```

Comment:
If you have an error in request - you should reach rejected case. In only this case you can not to have payload. Fulfillled - it successful case. You can’t not to have payload there. Probably this happens because you add if checks in thunks.